R/sidebar.R

```diff
@@ -138,28 +195,37 @@ layout_sidebar <- function(sidebar = sidebar(), ..., fillable = FALSE, fill = TR
 }
 
 sidebar_js_init <- function() {
-  tags$script("data-bslib-sidebar-init" = NA, HTML(
-    "
-    var thisScript = document.querySelector('script[data-bslib-sidebar-init]');
-    thisScript.removeAttribute('data-bslib-sidebar-init');
-
-    // If this layout is the innermost layout, then allow it to add CSS
-    // variables to it and its ancestors (counting how parent layouts there are)
-    var thisLayout = $(thisScript).parent();
-    var noChildLayouts = thisLayout.find('.bslib-sidebar-layout').length === 0;
-    if (noChildLayouts) {
-      var parentLayouts = thisLayout.parents('.bslib-sidebar-layout');
-      // .add() sorts the layouts in DOM order (i.e., innermost is last)
-      var layouts = thisLayout.add(parentLayouts);
-      var ctrs = {left: 0, right: 0};
-      layouts.each(function(i, x) {
-        $(x).css('--bslib-sidebar-counter', i);
-        var right = $(x).hasClass('sidebar-right');
-        $(x).css('--bslib-sidebar-overlap-counter', right ? ctrs.right : ctrs.left);
-        right ? ctrs.right++ : ctrs.left++;
-      });
+  tags$script("data-bslib-sidebar-init" = NA, HTML("
+(function() {
```

Comment:
Let's do this in a follow-up PR (since we also do this for cards, let's think about having a helper function to provide the "initialization" templating)

Reply:
Took out this change in 0e11047fe8fb1c7a9d794228d13807e5c84bdca6 for a cleaner diff and opened https://github.com/rstudio/bslib/issues/517